server/util/claims/claims.go

```diff
@@ -312,6 +316,11 @@ func AuthContextFromClaims(ctx context.Context, c *Claims, err error) context.Co
 		return authutil.AuthContextWithError(ctx, err)
 	}
 	ctx = context.WithValue(ctx, authutil.ContextTokenStringKey, tokenString)
+	return AuthContextWithClaims(ctx, c)
+}
+
+// Returns a Context containing auth state for the the provided Claims.
+func AuthContextWithClaims(ctx context.Context, c *Claims) context.Context {
```

Comment:
FromClaims and WithClaims are not the clearest names. Maybe we should rename AuthContextFromClaims to AuthContextWithJWT?

Reply:
Yeah, definitely agree. What do you think about the approach I just pushed here which gets rid of the new function and adds the JWT as a parameter to `AuthContextFromClaims` so the caller can provide it if it's known? I thought about checking for it in the context and not setting it if it already exists in the context, but I don't think that's actually the correct thing to do based on the calls in `oidc.go`.

Reply:
I don't love those bare empty string literals everywhere. Ideally someone reading the code shouldn't have to jump to the function declaration to know what a parameter is. IMO your initial approach was better but could have done with clearer naming.

Reply:
SG, updated per your original review!

Reply:
My last suggestion (sorry) is to call the second function simply AuthContext. My thinking is that 1) this function is already part of the `claims` package so the 'claims' suffix is not necessary and 2) `AuthContextWithJWT` includes Claims in the context as well whereas the name 'WithClaims' might suggest that only this other function does. This way `claims.AuthContextWithJWT` adds functionality (adding JWT) on top of `claims.AuthContext`.

Reply:
Yeah, I was wondering about calling it that too. Done. Thanks!